backend/app/routers/experiment_templates.py

```diff
@@ -134,23 +245,26 @@ def find_specific_experiment_templates(
         if pagination is not None
         else {}
     )
+    if len(search_query) > 0:
+        search_conditions.append(Text(search_query))
+
+    if user is None and only_mine:
```

Comment:
It would be much more clear if we change the conditions now
```python
if only_mine:
    if user is not None:
          ...
    else:
          raise ...
```